bin/supervise.sh

```diff
@@ -1,31 +1,44 @@
 #!/usr/bin/env bash
 
+timestamp()
+{
+    if [[ -z "$TIMESTAMP_MODE" ]]; then
+        if ! command -v date >/dev/null; then
+            echo 'supervise.sh requires `date`' >&2
+            exit 1
+        fi
+
+        if [[ "$(date +%3N)" =~ ^[0-9]{3}$ ]]; then
+            TIMESTAMP_MODE="date_ns"
+        elif command -v adjtimex >/dev/null && command -v awk >/dev/null; then
+            TIMESTAMP_MODE="adjtimex"
+        else
+            TIMESTAMP_MODE="date_s"
+        fi
+    fi
+
+    case "$TIMESTAMP_MODE" in
+        date_ns)
+            date -u +"%Y-%m-%dT%H:%M:%S.%3NZ"
+            ;;
+        adjtimex)
+            local timestamp_ns="$(adjtimex | awk '/(time.tv_sec|time.tv_usec)/ { printf("%06d", $2) }')"
+            local ms="${timestamp_ns: -6:3}"
+            local timestamp_s="${timestamp_ns::-6}"
```

Comment:
this is microseconds, not nanoseconds as far as I can tell (matching the output from `adjtimex`)

```suggestion
            local timestamp_us="$(adjtimex | awk '/(time.tv_sec|time.tv_usec)/ { printf("%06d", $2) }')"
            local ms="${timestamp_us: -6:3}"
            local timestamp_s="${timestamp_us::-6}"
```

Reply:
Good catch. I just got mixed up